corehq/apps/formplayer_api/smsforms/tests/test_formplayer_interface.py

```diff
@@ -0,0 +1,67 @@
+import uuid
+
+import requests_mock
+from django.conf import settings
+from django.test import TestCase
+
+from corehq.apps.formplayer_api.smsforms.api import FormplayerInterface
+from corehq.apps.formplayer_api.utils import get_formplayer_url
+from corehq.util.hmac_request import get_hmac_digest
+
+
+class TestFormplayerInterface(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.session_id = uuid.uuid4().hex
+        self.domain = uuid.uuid4().hex
+        self.user_id = uuid.uuid4().hex
+        self.interface = FormplayerInterface(self.session_id, self.domain, self.user_id)
+
+    def test_get_raw_instance(self):
+        expected_request_data = {
+            'action': 'get-instance',
+            'session-id': self.session_id,
+            'session_id': self.session_id,
+            'domain': self.domain,
+            'oneQuestionPerScreen': True,
+            'nav_mode': 'prompt',
+        }
+        self._test_request("get-instance", self.interface.get_raw_instance, expected_request_data, {})
```

Comment:
The intent as you've shown it is correct - I'll take another stab at cleaning it up.

> Specifically, I think it is probably not worth it to generate a digest in the tests in the exact same way that it is generated in the code.

I agree this isn't great but I don't think there's any value in having a constant here either since that will change with any changes to the input data and be very annoying to go and re-calculate (likely using the same function anyway). Perhaps I'll just verify that there's a non-blank header value.